### ISSUE TITLE
Need to validate new theme names

### ISSUE THREAD

**Issue by childsb** [maqetta/maqetta#263]
In the New Theme... dialog, you can type a new theme name that contains a
blank. A theme name with a space of course won't fly because we do a
substitution of <newthemename> for "claro" (or whatever the original theme name
is) in all of the CSS files.

In addition to preventing space characters, we should validate to prevent all
characters not allowed by CSS 'ident' construct. There has to be a regex that
can be used, but I don't know offhand how to create a regex that takes into
account all of the possible Unicode characters that might be possible. Probably
OK in short-term to just allow a range of ASCII chars, but need to create a new
bug then to allow globalized names.

@jmockler, @billreed63

bug: 7657

**Comment by JonFerraiolo** [comment]
Users run into this issue, so would be nice to fix as quickly as possible, but for M2 only if we have time and if we are absolutely sure the fix is safe. Might be the cause of #552

**Comment by childsb** [comment]
fixed with commit bf133d85175b8549f865d0b29c531c1aaed2406a

**Comment by JonFerraiolo** [comment]
@peller

Reopening, but pushing to M3. The critical part of this bug has been addressed: users cannot hang themselves anymore by typing a theme name that we can't deal with. However, the UI needs to be improved. Right now, we disable the OK button when the theme name is invalid. Instead, we should use Dojo's ValidationTextBox (or whatever) which will put an explanatory overlay message when the user has typed a value that is invalid, providing a hint as to why the value is invalid.

**Comment by JonFerraiolo** [comment]
@peller, @waynevicknair, @jhpedemonte

Also, the regular expression used is fine for now, but has at least one "bug" and should be improved to support users across the globe. Look at http://www.w3.org/TR/CSS21/syndata.html#characters and you'll see that identifiers (which is what we need because a classname is an identifier according to CSS parser) can contain [a-zA-Z0-9] and ISO 10646 characters U+00A0 and higher, plus the hyphen (-) and the underscore (_), and (this is the "bug") they cannot start with a digit, two hyphens, or a hyphen followed by a digit. Anyone want to volunteer to author such a regular expression?

**Comment by waynevicknair** [comment]
i eat regular expressions for breakfast; i'll take it. that one will require some testing tho. all those ISO characters need to be escaped and there are look behind assertions for hyphen followed by hyphen and hyphen followed by digit.

**Comment by childsb** [comment]
i heard the original name for alpha-bits was /[A-Z]+/, maybe thats what you've been eating?

thanks wayne, these complex regex's make my head hurt.

**Comment by waynevicknair** [comment]
checked-in NewTheme.js under commit 2a76cba804040f1b4afa3a72a973b2e166b3a49d with CSS specification compliant regular expression to validate theme names.

**Comment by waynevicknair** [comment]
looks like i missed the part about the ValidationTextBox. re-opening to address that issue.

**Comment by waynevicknair** [comment]
Added ValidationTextBox to NewTheme dialog with commit 6ff926d6142a7c6d57cd03bc1e4ba96b8263b4a4
